pycode/memilio-simulation/tools/generate_stubs.py

```diff
@@ -34,39 +37,113 @@
     file_path = os.path.dirname(os.path.abspath(__file__))
     package_dir = os.path.abspath(os.path.join(
         file_path, "../../memilio-simulation-stubs"))
-    output_dir = os.path.join(package_dir, "memilio-stubs/simulation")
-    output_module_dir = os.path.join(output_dir, 'memilio')
 
+    # delete stubs if they already exist
+    try:
+        shutil.rmtree(os.path.join(package_dir, "memilio-stubs"))
+    except:
+        pass
     # create folders, if they do not exist
     try:
-        # output_module_dir is generated by pybind11_stubgen
-        os.makedirs(output_dir)
+        os.makedirs(package_dir)
     except:
         pass
 
-    # get all model modules from memilio.simulation
-    # if package structure changes this needs to be adjusted
-    models = [m.name for m in pkgutil.iter_modules(
-        memilio.simulation.__path__)]
-
-    # generate stubs and moce them into correct folder with right name
     # memilio-stubs/simulation module needs same structure as memilio/simulation
+    # test --include-docstrings, --doc-dir PATH for better docs
     subprocess.check_call(
-        [python_interpreter, '-m', 'pybind11_stubgen', '--ignore-all-errors', '-o', output_dir, 'memilio._simulation'])
-    os.rename(os.path.join(output_module_dir, '_simulation.pyi'),
-              os.path.join(output_dir, '__init__.pyi'))
+        ['stubgen', '--include-docstrings', '-o', package_dir, '-p', 'memilio.simulation'])
+
+    # TODO: several error in the stubs (output of running mypy on codebase with stubs generated by mypy, need to set flag_configure_generated_stubs=False)
+    #   - fix numpy.float64[m, 1] to numpy.float64[M, Literal(1)] with M = typing.TypeVar("M", bound=int) at top of file
+    #   - or numpy.ndarray[numpy.float64[m, 1]] to numpy.ndarray[tuple[M, typing.Literal[1]], numpy.dtype[numpy.float64]]
+    #   - also numpy.ndarray[numpy.float64[m, n], flags.f_contiguous] needs to get n defined
+    #   - change following error:
```

Comment:
I don't understand what this comment is supposed to tell me, at least the part in parantheses.

Also, is that something that requires a new issue?

Reply:
Improved the comment to be more comprehensive and moved parts of it to a different place. Those **TODOs* are low priority and need enough time to implement. I would move them into a future issue.